src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2789,6 +2789,37 @@ protected override IPointerTypeSymbol CommonCreatePointerTypeSymbol(ITypeSymbol
             return CreatePointerTypeSymbol(elementType.EnsureCSharpSymbolOrNull<ITypeSymbol, TypeSymbol>("elementType"));
         }
 
+        protected override INamedTypeSymbol CommonCreateTupleTypeSymbol(ITypeSymbol[] elementTypes, string[] elementNames)
```

Comment:
The closest analog to this would be INamedTypeSymbol.Construct.  So follow whatever that does.